data/data/bootstrap/baremetal/files/usr/local/bin/startironic.sh.template

```diff
@@ -0,0 +1,111 @@
+# nmcli show provisioning there instead, but we need to confirm that works with the static-ip-manager
+while [ -z "$(ip -4 address show dev "$PROVISIONING_NIC" | grep -oP '(?<=inet\s)\d+(\.\d+){3}' | head -n 1)" ]; do
+    sleep 1
+done
+
+# set password for mariadb
+mariadb_password=$(echo $(date;hostname)|sha256sum |cut -c-20)
+
+IRONIC_SHARED_VOLUME="ironic"
+podman volume create $IRONIC_SHARED_VOLUME || true
+
+# Apparently network-online doesn't necessarily mean iptables is ready, so wait until it is..
+while ! iptables -L; do
+  sleep 1
+done
+
+# Add firewall rules to ensure the IPA ramdisk can reach httpd, Ironic and the Inspector API on the host
+for port in 80 5050 6385 ; do
+    if ! sudo iptables -C INPUT -i $PROVISIONING_NIC -p tcp -m tcp --dport $port -j ACCEPT > /dev/null 2>&1; then
+        sudo iptables -I INPUT -i $PROVISIONING_NIC -p tcp -m tcp --dport $port -j ACCEPT
+    fi
+done
+
+# Start dnsmasq, http, mariadb, and ironic containers using same image
+# Currently we do this outside of a pod because we need to ensure the images
+# are downloaded before starting the API pods
+podman run -d --net host --privileged --name mariadb \
+     -v $IRONIC_SHARED_VOLUME:/shared:z --entrypoint /bin/runmariadb \
+     --env MARIADB_PASSWORD=$mariadb_password ${IRONIC_IMAGE}
+
+podman run -d --net host --privileged --name dnsmasq \
+     --env PROVISIONING_INTERFACE=$PROVISIONING_NIC \
+     -v $IRONIC_SHARED_VOLUME:/shared:z --entrypoint /bin/rundnsmasq ${IRONIC_IMAGE}
+
+podman run -d --net host --privileged --name httpd \
+     --env PROVISIONING_INTERFACE=$PROVISIONING_NIC \
+     -v $IRONIC_SHARED_VOLUME:/shared:z --entrypoint /bin/runhttpd ${IRONIC_IMAGE}
+
+# Set CACHEURL to the default route, so we try to consume any images cached on the host
+# running the VM (dev-scripts configures a cache here), if none is found then the
+# downloader containers just skip and download from the internet location
+CACHEURL="http://$(ip r | grep default | head -n1 | awk '{print $3}')/images"
+podman run -d --net host --name ipa-downloader \
+     --env CACHEURL=${CACHEURL} \
+     -v $IRONIC_SHARED_VOLUME:/shared:z ${IPA_DOWNLOADER_IMAGE} /usr/local/bin/get-resource.sh
+
+podman run -d --net host --name coreos-downloader \
+     --env CACHEURL=${CACHEURL} \
+     -v $IRONIC_SHARED_VOLUME:/shared:z ${COREOS_DOWNLOADER_IMAGE} /usr/local/bin/get-resource.sh $RHCOS_IMAGE_BASE_URL
+
+# Start the conductor so db sync can happen while images download
+sudo podman run -d --net host --privileged --name ironic-conductor \
+     --env MARIADB_PASSWORD=$mariadb_password \
+     --env PROVISIONING_INTERFACE=$PROVISIONING_NIC \
+     --env OS_CONDUCTOR__HEARTBEAT_TIMEOUT=120 \
+     --entrypoint /bin/runironic-conductor \
+     -v $IRONIC_SHARED_VOLUME:/shared:z ${IRONIC_IMAGE}
+
+# Wait for images to be downloaded/ready
+while ! curl --fail http://localhost:80/images/rhcos-ootpa-latest.qcow2.md5sum ; do sleep 1 ; done
```

Comment:
I suppose we could detect a failure a bit quicker here if we watched the state of the `coreos-downloader` container, and then did a one time check once the downloader has exited.

Reply:
Ah yeah that's a good idea, thanks!

Reply:
looks like `podman wait` will help with this.